Automattic/wp-calypso
# Update/cancelation modal

#### P2
[ pebzTe-6r-p2](pebzTe-oH-p2)

#### GitHub Issue
[1039-gh-Automattic/martech](https://href.li/?https://github.com/Automattic/martech/issues/1039)

#### Description
This PR covers the points 1. and 2. from the P2 post plan:
1. Add a feature flag, e.g. purchases/cancellation-modal so we can deploy the individual component safely.
2. The skeleton flow that is behind a feature flag. It only needs to show placeholder content and doesn’t even need to perform the real actions.

#### Feature flags:
URL query: `?flags=pre-cancellation-modal`
Shell command: `ENABLE_FEATURES=pre-cancellation-modal yarn start`

#### Test plan:
1. Access Calypso enabling the feature `pre-cancellation-modal`
2. Access the purchase page
4. Verify the skeleton flow for the Pre Cancellation Modal behaves as expected.
5. Verify the only "disjoint case" is for Subscription in the refund window, having a custom domain.

#### Notes:
- To simplify the review and test of this PR it does not yet contain points 3,4,5,6,7 that will be treated in separated PRs.
- The Pre Cancellation Modal currently contains some *debugs values* to help debug the Skeleton Flow better <img width="547" alt="Screenshot 2022-10-25 at 15 21 51" src="https://user-images.githubusercontent.com/5706607/197785410-94d79bb4-68b1-466b-af00-d02fb46a115a.png">
- The refund text and the contact support link were been ported to the cancellation modal <img width="541" alt="Screenshot 2022-10-25 at 15 21 11" src="https://user-images.githubusercontent.com/5706607/197785377-f5e52b3f-0f51-45fd-a5d4-53bb55dbee44.png">



#### Before:
![image](https://user-images.githubusercontent.com/5706607/197782203-4f70698c-11fa-4766-afba-bbc403c134bb.png)

#### After:
https://user-images.githubusercontent.com/5706607/197785291-6f906031-7f8a-4b9c-8efb-567b49e23f4f.mp4
